### PR TITLE
Setup simple server on gateway

### PR DESCRIPTION
**Pull request by evan-schott**
Simple net/http server

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (lthibault) [inline .vscode/launch.json]: You probably don't want to commit anything under `.vscode/*`

File: main.go [hunk 2]
```diff
@@ -106,11 +109,75 @@ func runGateway(c *cli.Context, n *server.Node) error {
 	var ch csp.Chan                         // TODO:  csp.NewChan()
 	n.Vat.Export(chanCap, chanProvider{ch}) // TODO: Will this also be exported to peers we find in the future?
 
-	return errors.New("NOT IMPLEMENTED")
+	log2.Println("starting server, listening on port " + getServerPort())
```
Comment (lthibault) [inline main.go]: Just use `logger.Info(...)` instead of importing a separate logging library.

File: main.go [hunk 3]
```diff
@@ -106,11 +109,75 @@ func runGateway(c *cli.Context, n *server.Node) error {
...
+	http.HandleFunc("/slight-echo", SlightEchoHandler)
+
+	return http.ListenAndServe(":"+getServerPort(), nil) // Can test with: curl -X GET -H "Content-Type: application/json" -d '{"message": "Hello, World!"}' http://localhost:8080/slight-echo
+}
+
+// DefaultPort is the default port to use if once is not specified by the SERVER_PORT environment variable
+const DefaultPort = "8080"
+
+// From https://github.com/aautar/go-http-echo/blob/master/echo.go#L0-L20
+func getServerPort() string {
```
Comment (lthibault) [inline main.go]: This is overkill.  I would just pass `":8080"` to `ListenAndServe` directly.

**Comment by evan-schott** [comment]
@lthibault

File: main.go [hunk 4]
```diff
@@ -127,7 +194,9 @@ func waitPeers(c *cli.Context, n *server.Node) (bool, error) {
 		it, release := n.View().Iter(ctx, queryAll())
 		defer release()
 
+		counter := 0
```
Comment (lthibault) [inline main.go]: What is this for?

**Review by lthibault — CHANGES_REQUESTED** [review]

File: main.go [hunk 5]
```diff
@@ -106,11 +108,62 @@ func runGateway(c *cli.Context, n *server.Node) error {
...
+	return http.ListenAndServe(":8080", nil) // Can test with: curl -X GET -H "Content-Type: application/json" -d '{"message": "Hello, World!"}' http://localhost:8080/slight-echo
+}
+
+// EchoHandler echos back the request as a response
+// From https://github.com/aautar/go-http-echo/blob/master/echo.go#L21-L40
+func EchoHandler(writer http.ResponseWriter, request *http.Request) {
+
+	logger.Info("Echoing back request made to " + request.URL.Path + " to client (" + request.RemoteAddr + ")")
+
+	writer.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (lthibault) [inline main.go]: Setting headers is unnecessary.  I'd remove this from both handlers, just to keep the visual noise down.

**Review by lthibault — CHANGES_REQUESTED** [review]

File: main.go [hunk 6]
```diff
@@ -106,11 +108,62 @@ func runGateway(c *cli.Context, n *server.Node) error {
...
+	request.Write(writer)
+}
+
+func SlightEchoHandler(writer http.ResponseWriter, request *http.Request) {
+	logger.Info("Slightly echoing back request made to " + request.URL.Path + " to client (" + request.RemoteAddr + ")")
+
+	writer.Header().Set("Access-Control-Allow-Origin", "*")
+	writer.Header().Set("Access-Control-Allow-Headers", "Content-Range, Content-Disposition, Content-Type, ETag")
+
+	var payload map[string]interface{}
```
Comment (lthibault) [inline main.go]: Recommend defining a struct for this.  Maps are a PITA to use with JSON.

**Review by lthibault — CHANGES_REQUESTED** [review]

File: main.go [hunk 7]
```diff
@@ -122,35 +122,34 @@ func EchoHandler(writer http.ResponseWriter, request *http.Request) {
...
-func SlightEchoHandler(writer http.ResponseWriter, request *http.Request) {
-	logger.Info("Slightly echoing back request made to " + request.URL.Path + " to client (" + request.RemoteAddr + ")")
+type Payload struct {
+	Message string `json:"message"`
+}
 
-	writer.Header().Set("Access-Control-Allow-Origin", "*")
-	writer.Header().Set("Access-Control-Allow-Headers", "Content-Range, Content-Disposition, Content-Type, ETag")
+func SlightEchoHandler(writer http.ResponseWriter, request *http.Request) {
+	log.Info("Slightly echoing back request made to " + request.URL.Path + " to client (" + request.RemoteAddr + ")")
```
Comment (lthibault) [inline main.go:136]: Just FYI:  you can use `logger.Infof("slightly echoing back request made to %s to client", request.URL.Path)`.

**Review by lthibault — APPROVED** [review]
LGTM 👍

**Pull request merged**